staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1573,22 +1574,49 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 }
 
 // startObservingCount starts monitoring given prefix and periodically updating metrics. It returns a function to stop collection.
-func (e *Store) startObservingCount(period time.Duration, objectCountTracker flowcontrolrequest.StorageObjectCountTracker) func() {
+func (e *Store) startObservingCount(period time.Duration, objectCountTracker flowcontrolrequest.StorageObjectCountTracker, isNamespaced bool) func() {
 	prefix := e.KeyRootFunc(genericapirequest.NewContext())
 	resourceName := e.DefaultQualifiedResource.String()
 	klog.V(2).InfoS("Monitoring resource count at path", "resource", resourceName, "path", "<storage-prefix>/"+prefix)
 	stopCh := make(chan struct{})
 	go wait.JitterUntil(func() {
 		count, err := e.Storage.Count(prefix)
 		if err != nil {
-			klog.V(5).InfoS("Failed to update storage count metric", "err", err)
+			klog.V(5).InfoS("Failed to get storage count for %s: %v", "err", resourceName, err)
 			count = -1
 		}
 
+		klog.Infof("[debug] Object count from storage for resource %q: %d", resourceName, count)
 		metrics.UpdateObjectCount(resourceName, count)
 		if objectCountTracker != nil {
 			objectCountTracker.Set(resourceName, count)
 		}
+
+		if !isNamespaced {
+			return
+		}
+
+		namespaces := &corev1.NamespaceList{}
+		err = e.Storage.GetList(context.TODO(), "/namespaces", storage.ListOptions{Predicate: storage.Everything, Recursive: true}, namespaces)
```

Comment:
This is the part seems a little hacky but I wasn't sure how else to get namespaces at this layer. Open to better suggestions.